### PR TITLE
Mkw ihme handoff

### PR DESCRIPTION
**Pull request by mwalte10**
This is about 95% of the way there, just need to confirm with Avenir whether a few things have been implemented. 

In the mean time, I would appreciate feedback on:

- Formatting and spacing: I've tried to autoformat but unsure if it has worked. Any suggestions on how to make this easier welcome! 
- Ordering + function nesting: Please let me know if I should pull out any of the functions that contain multiple other processes (specifically calc_art_initates)
- Variable naming
- Any ways that I can make the code easier to read (is there away to avoid the intermediate.children. prefix?)
- The logic in calc_art_last_year. It's a complicated function as it has multiple if's, may be easiest to talk through this?

### CODE REVIEW

**Review by r-ash — CHANGES_REQUESTED** [review]
This looks good from a quick look. Looks like accidentally copied a few of the package files into the tests dir? `tests` has a `DESCRIPTION`, `LICENSE`, etc. and an `R` directory in there. Can we remove that?

There is also some test data the root level `project_parameters_child.rds` which I think shouldn't be there.

Few comments below but I still have to look into a couple of your questions.

Couple of things we should do before we merge this
* Get main merged into this branch, I can do this as I've made some changes to that which I think will cause some clashes here
* Tidy up the test data scripts so that they work on anyones machine, atm these have some hardcoded references to your directory structure

> Formatting and spacing: I've tried to autoformat but unsure if it has worked. Any suggestions on how to make this easier welcome!
Did you autoformat using `styler`? from Rstudio, "Addins -> Style active file". I've not used that myself, but might be what we need. In general this can be a bit of a pain as what styler does might be different to what codefactor is warning you about. Which will probably be different to what someone using a different IDE will want from the formatting. We probably just need to think this through and set some configuration. Let me know what you used for the formatting, I'll put it on my list to look into this.

> Ordering + function nesting: Please let me know if I should pull out any of the functions that contain multiple other processes (specifically calc_art_initates)
Still to look at this

> Variable naming
Still to look at this

> Any ways that I can make the code easier to read (is there away to avoid the intermediate.children. prefix?)
Yes you can pull that out at the top of a function with `const auto ic = intermediate.children` will make a the prefix a bit more manageable. Or use any name you want.

> The logic in calc_art_last_year. It's a complicated function as it has multiple if's, may be easiest to talk through this?
Still to look at this

File: inst/include/child_model_simulation.hpp [hunk 1]
```diff
@@ -731,16 +1422,37 @@ void run_child_model_simulation(int time_step,
...
-  // !!!TODO: also need to fix the looping order for some loops
-  // !!!TODO: put this in an if statement to only run if the first year of ART has passed
-  internal::run_child_art_initiation(time_step, pars, state_curr, state_next, intermediate);
-  internal::run_child_art_mortality(time_step, pars, state_curr, state_next, intermediate);
+
+  if(cpars.mat_prev_input(time_step)){
+    internal::run_wlhiv_births_input_mat_prev(time_step, pars, state_curr, state_next, intermediate);
+  }else{
+    internal::run_wlhiv_births(time_step, pars, state_curr, state_next, intermediate);
+  }
```
Comment (r-ash) [inline inst/include/child_model_simulation.hpp:1433]: Nit picky style comment
```suggestion
  if (cpars.mat_prev_input(time_step)) {
    internal::run_wlhiv_births_input_mat_prev(time_step, pars, state_curr, state_next, intermediate);
  } else {
    internal::run_wlhiv_births(time_step, pars, state_curr, state_next, intermediate);
  }
```

File: inst/include/child_model_simulation.hpp [hunk 2]
```diff
@@ -731,16 +1422,37 @@ void run_child_model_simulation(int time_step,
...
+    internal::ctx_need_cov(time_step, pars, state_curr, state_next, intermediate);
+    internal::run_child_natural_history(time_step, pars, state_curr, state_next, intermediate);
+    internal::run_child_hiv_mort(time_step, pars, state_curr, state_next, intermediate);
+    internal::add_child_grad(time_step, pars, state_curr, state_next, intermediate);
+
+    //assume paed art doesn't start before adult
+    if(time_step > pars.base.options.ts_art_start){
+      internal::hc_art_initiation_by_age(time_step, pars, state_curr, state_next, intermediate);
+      //mortality among those on ART less than one year
+      internal::onART_mortality(time_step, pars, state_curr, state_next, intermediate, 0);
```
Comment (r-ash) [inline inst/include/child_model_simulation.hpp:1446]: Another stle comment but I'd prefer to keep this consistent `on_art_mortality`

File: inst/include/child_model_simulation.hpp [hunk 3]
```diff
@@ -67,660 +68,1350 @@ void run_child_ageing(int time_step,
...
+      for (int ht = 1; ht < ss.hTS; ++ht) {
+        intermediate.children.df += cpars.local_adj_factor * cpars.fert_mult_onart(a) * ((state_next.base.h_art_adult(ht, hd, a, 1) + state_curr.base.h_art_adult(ht, hd, a, 1)) / 2);
+      } //end hTS
+    } // end hDS
+
+
+    if (intermediate.children.nHIVcurr > 0) {
+      intermediate.children.df = intermediate.children.df / ((intermediate.children.nHIVcurr + intermediate.children.nHIVlast) / 2);
+    }else{
+      intermediate.children.df = 1;
```
Comment (r-ash) [inline inst/include/child_model_simulation.hpp:118]: I think unpacking the `intermediate.children` will simplify this
```suggestion
  const auto ic = intermediate.children;
  ic.births_sum = state_next.base.births;

  for (int a = 0; a < pars.base.options.p_fertility_age_groups; ++a) {
    ic.nHIVcurr = 0.0;
    ic.nHIVlast = 0.0;
    ic.df = 0.0;

    for (int hd = 0; hd < ss.hDS; ++hd) {
      ic.nHIVcurr += state_next.base.h_hiv_adult(hd, a, 1);
      ic.nHIVlast += state_curr.base.h_hiv_adult(hd, a, 1);
      for (int ht = 0; ht < ss.hTS; ++ht) {
        ic.nHIVcurr += state_next.base.h_art_adult(ht, hd, a, 1);
        ic.nHIVlast += state_curr.base.h_art_adult(ht, hd, a, 1);
      } //end hTS
    } //end hDS

    ic.prev = ic.nHIVcurr / state_next.base.p_total_pop(a + 15,1);

    for (int hd = 0; hd < ss.hDS; ++hd) {
      ic.df += cpars.local_adj_factor * cpars.fert_mult_by_age(a) * cpars.fert_mult_offart(hd) * ((state_next.base.h_hiv_adult(hd, a, 1) + state_curr.base.h_hiv_adult(hd, a, 1)) / 2);
      //women on ART less than 6 months use the off art fertility multiplier
      ic.df += cpars.local_adj_factor * cpars.fert_mult_by_age(a) * cpars.fert_mult_offart(hd) * ((state_next.base.h_art_adult(0, hd, a, 1) + state_curr.base.h_art_adult(0, hd, a, 1)) / 2);
      for (int ht = 1; ht < ss.hTS; ++ht) {
        ic.df += cpars.local_adj_factor * cpars.fert_mult_onart(a) * ((state_next.base.h_art_adult(ht, hd, a, 1) + state_curr.base.h_art_adult(ht, hd, a, 1)) / 2);
      } //end hTS
    } // end hDS


    if (ic.nHIVcurr > 0) {
      ic.df = ic.df / ((ic.nHIVcurr + ic.nHIVlast) / 2);
    }else{
      ic.df = 1;
```

File: inst/include/child_model_simulation.hpp [hunk 4]
```diff
@@ -67,660 +68,1350 @@ void run_child_ageing(int time_step,
...
+
+  for(int s = 0; s < ss.NS; ++s){
+    for(int a = 0; a < ss.hAG; ++a){
+      intermediate.children.p_hiv_neg_pop(a, s) = demog.base_pop(a, s) - state_next.base.p_hiv_pop(a, s);
+    }// end a
+  }//end s
+}
+
+template<typename ModelVariant, typename real_type>
+void convert_PMTCT_num_to_perc(int time_step,
```
Comment (r-ash) [inline inst/include/child_model_simulation.hpp:174]: I'd also keep the casing consistent here `convert_pmtct_num_to_perc`

File: inst/include/child_model_simulation.hpp [hunk 5]
```diff
@@ -67,660 +68,1350 @@ void run_child_ageing(int time_step,
...
+void convert_PMTCT_pre_bf(int time_step,
+                          const Parameters<ModelVariant, real_type> &pars,
+                          const State<ModelVariant, real_type> &state_curr,
+                          State<ModelVariant, real_type> &state_next,
+                          IntermediateData<ModelVariant, real_type> &intermediate) {
+  const auto cpars = pars.children.children;
   static_assert(ModelVariant::run_child_model,
-                "run_hiv_child_infections can only be called for model variants where run_child_model is true");
-  constexpr auto ss = StateSpace<ModelVariant>().base;
+                "convert_PMTCT_num_to_perc can only be called for model variants where run_child_model is true");
```
Comment (r-ash) [inline inst/include/child_model_simulation.hpp:224]:
```suggestion
                "convert_PMTCT_pre_bf can only be called for model variants where run_child_model is true");
```

File: inst/cpp_generation/model_input.csv [hunk 6]
```diff
@@ -28,11 +28,32 @@ paed_cd4_prog,hc1_cd4_prog,real_type,ModelVariant::run_child_model,,FALSE,1,chil
...
+paed_art_elig_age,hc_art_elig_age,int,ModelVariant::run_child_model,,FALSE,1,proj_years,,,
 paed_art_elig_cd4,hc_art_elig_cd4,real_type,ModelVariant::run_child_model,,TRUE,2,options.p_idx_hiv_first_adult,proj_years,,
 mort_art_rr,hc_art_mort_rr,real_type,ModelVariant::run_child_model,,FALSE,3,base.hTS,options.p_idx_hiv_first_adult,proj_years,
 paed_art_mort,hc1_art_mort,real_type,ModelVariant::run_child_model,,FALSE,3,children.hc1DS,base.hTS,children.hc1AG,
 adol_art_mort,hc2_art_mort,real_type,ModelVariant::run_child_model,,FALSE,3,children.hc2DS,base.hTS,children.hc2AG,
 artpaeds_isperc,hc_art_isperc,int,ModelVariant::run_child_model,,FALSE,1,proj_years,,,
-paed_art_val,hc_art_val,real_type,ModelVariant::run_child_model,,FALSE,1,proj_years,,,
+paed_art_val,hc_art_val,real_type,ModelVariant::run_child_model,,FALSE,2,4,proj_years,,
 init_art_dist,hc_art_init_dist,real_type,ModelVariant::run_child_model,,FALSE,2,options.p_idx_hiv_first_adult,proj_years,,
+adult_cd4_dist,adult_cd4_dist,real_type,ModelVariant::run_child_model,,FALSE,2,base.hDS,children.hc2DS,,
```
Comment (r-ash) [inline inst/cpp_generation/model_input.csv:39]: re variable naming, in the README there is a section "Leapfrog to frogger glossary" which has some naming rules. We added a set of prefixes there. I think fine if this is a new prefix for `adult` and `fert`. If so can we add to the README? But if not then I think let's try to move these to use those standardised prefixes?

Comment (r-ash) [reply inst/cpp_generation/model_input.csv:39]: That said - up to you. I don't feel super strongly about this if you think your names are easier to read and understand.

File: inst/include/child_model_simulation.hpp (hunk 2 above)
Comment (mwalte10) [reply inst/include/child_model_simulation.hpp]: changed!

File: inst/include/child_model_simulation.hpp (hunk 1 above)
Comment (mwalte10) [reply inst/include/child_model_simulation.hpp]: Changed in all instances.

**Pull request merged**